abbyy_to_epub3/parse_abbyy.py

```diff
@@ -203,286 +269,239 @@ def parse_abbyy(self):
             elem.tag == "page"
         ):
-            self.pages.append(deepcopy(elem))
-        else:
-            return
+            d = {'page_no': self.page_no}
 
-        # only garbage collect if we have found & processed a node
-        elem.clear()
-
-    def parse_content(self):
-        """ Parse each page of the book.  """
-        page_no = 0
-        d = {'page_no': page_no}
-
-        self.pages[0].clear()    # clear the memory first
-        for page in self.pages:
-            pagewidth = page.get('width')
-            pageheight = page.get('height')
-            block_per_page = page.getchildren()
+            self.pagewidth = elem.get('width')
+            self.pageheight = elem.get('height')
+            block_per_page = elem.iterchildren()
             if not block_per_page:
-                page_no += 1
-                continue
-            newpage = True
+                self.page_no += 1
+                return
+            self.newpage = True
 
+            # Most pages have multiple `<block>` elements
             for block in block_per_page:
-                blockattr = block.attrib
-                blockattr['pagewidth'] = pagewidth
-                blockattr['pageheight'] = pageheight
-                if self.is_block_type(block, "Text"):
-                    paras = block.findall(".//a:par", namespaces=self.nsm)
-                    # Some blocks can have multiple styles in them. We'll treat
-                    # those as multiple blocks.
-                    for para in paras:
-                        # Get the paragraph style and text
-                        para_id = para.get("style")
-                        if para_id not in self.paragraphs:
-                            self.logger.info(
-                                'Block {} has no paragraphStyle'.format(
-                                    para_id
-                                )
-                            )
-                            self.paragraphs[para_id] = dict()
-                        text = gettext(para).strip()
+                self.parse_block(block)
 
-                        # Ignore whitespace-only pars
-                        if not text:
-                            continue
+            # Mark up the last text block on the page, if there is one
+            add_last_text(self.blocks, self.page_no)
 
-                        # Get the paragraph role
-                        # FR6 docs have no structure, styles, roles
-                        if self.version == "FR10":
-                            role = self.paragraphs[para_id]['role']
-                        else:
-                            role = "FR6"
+            # For accessibility, create a page number at the end of every page
+            if self.metadata['PAGES_SUPPORT']:
+                d = {
+                    'type': 'Page',
+                    'text': self.page_no,
+                }
+                self.blocks.append(d)
```

Comment:
Is there a reason we're not just appending:

```
self.blocks.append({
  'type': 'Page',
  'text': self.page_no,
})
```
?

Is there a reason we set `d = dict()` after, even when it doesn't appear to be used after that point?